### ISSUE TITLE
err becomes NaN

### ISSUE THREAD

**Issue by skrealworld** [facebook/NAMAS#21]
Hi Thanks for this helpful framework. 
I am training this for longer document and summary pairs. After 2 epochs this err becomes nan.
local input, target = valid_data:next_batch(offset)
      local out = self.mlp:forward(input)
      local **err** = self.criterion:forward(out, target) * target:size(1)
 
After 2 epoch it is displaying me this: 
err     nan
NaN
{
  1 : CudaTensor - size: 64x291
  2 : CudaTensor - size: 64x291
  3 : CudaTensor - size: 64x5
}
err     nan
NaN
{
  1 : CudaTensor - size: 64x291
  2 : CudaTensor - size: 64x291
  3 : CudaTensor - size: 64x5
}
err     nan
NaN
{
  1 : CudaTensor - size: 64x291
  2 : CudaTensor - size: 64x291
  3 : CudaTensor - size: 64x5
}
[EPOCH : 20 LOSS: 0.000000 TOTAL: 850461 BATCHES: 13683]

Any help would be great. 
Thanks.

**Comment by skrealworld** [comment]
Nevermind, it was learning rate issue.

**Comment by rrkarim** [comment]
@skrealworld hey, what was the error? How did you solve it?

**Comment by rrkarim** [comment]
Decreasing the learning rate helps to alleviate the problem. Still, after 15 epochs err becomes zero.
